dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityInstanceQueryParams.java

```diff
@@ -734,6 +740,15 @@ public boolean hasUser()
         return user != null;
     }
 
+    /**
+     * Indicates whether this parameters specifies if tei is a potential
+     * duplicate. It can be true or false.
```

Comment:
It can be true or false is implicit in the return value and it can be removed from the comment.
What is confusing me here is the `has` vs `is`. The name of the method is `hasPotentialDuplicate` while in the comment it says that tei `is a potential duplicate`.
I think here this method is not telling you anything about the tei but it is telling if the parameters specify a potential duplicate filter.

Reply:
> I think here this method is not telling you anything about the tei but it is telling if the parameters specify a potential duplicate filter.

Isn't that the same point about the query parameter `potentialDuplicate` above? To me the query parameter does not convey a clear meaning.

Reply:
`hasPotentialDuplicate` is about the parameter in the class. So I'd leave the method name like that and explain it in the doc correctly as you are pointing out it is not correct

Reply:
IMHO the query parameter and all related fields should be called `includePotentialDuplicates`. To me this clearly says what it will do all the way down to the DB. Not much docs needed then 😋

Reply:
I think the include word is here used with a different meaning.
We have `includeDeleted` that if false by default and if true it will include all the deleted teis.
Here the `potentialDuplicate` is used to filter the result, and it is not a binary value, it can also be null.
I am not sure what it the best name here but if we put `includePotentialDuplicates` I would expect that if it is false, I would get all the teis but the ones that have potential duplicate equals true and if it is true I would get all the teis, also the ones with potential duplicate equals false

Reply:
Mhm, ok got it 👌🏻 `includePotentialDuplicates` is not good then.

Reply:
I think the name and comment is a bit confusing, although I understand the intention in the context of the whole file.

I would suggest improving the naming and comments as follows:
Name: "hasPotentialDuplicateFilter" or similar. Currently, the name confuses me with the actual property of the tei.

Comment: Describe it more like "Checks if the Parameters include potentialDuplicate as a filter. If the property is null, it implies we do not care whether a tei is a potentialDuplicate or not."